src/data_snack/snack.py

```diff
@@ -20,16 +17,17 @@ class Snack:
 
     connection: Connection
     registry: Dict[Text, EntityRegistry] = field(default_factory=dict)
+    key_factory: KeyFactory = field(default=key_factory)
```

Comment:
Should we test snack for cluster_key_factory?

Reply:
Yeah, I will add test to check if we change the `key_factory` will the generated key will change.